vault/acl.go

```diff
@@ -291,8 +291,8 @@ func (a *ACL) Capabilities(ctx context.Context, path string) (pathCapabilities [
 }
 
 // AllowOperation is used to check if the given operation is permitted.
-func (a *ACL) AllowOperation(ctx context.Context, req *logical.Request, capCheckOnly bool) (ret *ACLResults) {
-	ret = new(ACLResults)
+func (a *ACL) AllowOperation(ctx context.Context, req *logical.Request, capCheckOnly bool) (ret ACLResults) {
+	ret = ACLResults{}
```

Comment:
Benchmarks showed that this eliminated an allocation and thereby gave a significant performance improvement.  I didn't want to introduce a performance regression when handling large "directories".

Reply:
How significant?

Reply:
With my change:

BenchmarkHasNonDenyCapability-8   	20000000	        69.5 ns/op	       0 B/op	       0 allocs/op

Without:

BenchmarkHasNonDenyCapability-8   	20000000	       117 ns/op	      48 B/op	       1 allocs/op

Reply:
Can't this line be removed entirely? `ret` is already initialized.

Reply:
> Can't this line be removed entirely? `ret` is already zero initialized.

Indeed it can, and I will remove it if we keep my change from struct pointer to struct.